### PR TITLE
v1.13 backports 2023-04-11

### PR DESCRIPTION
**Pull request by pchaigno**
- [x] #24521 -- gha: Skip HTTPRouteListenerHostnameMatching test temporarily (@sayboras)
     - Trivial conflict.
 - [x] #24742 -- pkg/service: Extend unit test cases (@aditighag)
 - [x] #24612 -- bgpv1: expose bgp peering state via cilium agent cli (@harsimran-pabla)
     - Some compilation errors due to silent conflict.
     - Needed to remove Family_SAFI_MUP.
 - [x] #24666 -- helm: mandate issuer configuration when using cert-manager to generate certificates (@giorio94)
     - Trivial conflict.
 - [x] #24769 -- loader: Don't compile `.asm` files by default (@pchaigno)
 - [x] #24792 -- bpf: dsr: fix parsing of IPv6 AUTH extension header (@julianwiedmann)
 - [x] #24797 -- bpf: policy: fix handling of ICMPv6 packet with extension headers (@julianwiedmann)
 - [x] #24773 -- ipsec: Clean up stale XFRM policies and states (@pchaigno)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 24521 24742 24612 24666 24769 24792 24797 24773; do contrib/backporting/set-labels.py $pr done 1.13; done
```

### CODE REVIEW

**Comment by pchaigno** [comment]
/test-backport-1.13

File: .github/workflows/conformance-gateway-api.yaml [hunk 1]
```diff
@@ -154,8 +154,9 @@ jobs:
           spec:
             controllerName: io.cilium/gateway-controller
           EOF
-          
-          GATEWAY_API_CONFORMANCE_TESTS=1 go test -v ./operator/pkg/gateway-api --gateway-class cilium --debug -test.run "TestConformance"
+
+          GATEWAY_API_CONFORMANCE_TESTS=1 go test -v ./operator/pkg/gateway-api --gateway-class cilium --debug -test.run "TestConformance" \
```
Comment (sayboras) [inline .github/workflows/conformance-gateway-api.yaml:158]: seems like the upstream conformance framework didn't accept -test.skip flag, we can try with the below expression instead

```
GATEWAY_API_CONFORMANCE_TESTS=1 go test -v ./operator/pkg/gateway-api --gateway-class cilium --debug -test.run "TestConformance" -test.run '!TestConformance/HTTPRouteListenerHostnameMatching'
```
```

**Review by julianwiedmann — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]
Thanks!

**Comment by pchaigno** [comment]
/test-backport-1.13

**Review by sayboras — APPROVED** [review]
Thanks and LGTM :heavy_check_mark:

**Comment by harsimran-pabla** [comment]
Thank you, LGTM

**Review by harsimran-pabla — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]

**Review by aditighag — APPROVED** [review]
Looks good for my backports. Thanks!

**Pull request merged**